File: LeetCode/Sorting/1200-Minimum-Absolute-Difference.py
"""
Problem: 1200. Minimum Absolute Difference
Platform: LeetCode
Difficulty: Easy

Approach:
1. Sort the array.
2. Find the minimum difference between adjacent elements.
3. Collect all adjacent pairs having that minimum difference.

Time Complexity: O(n log n)
Space Complexity: O(1) (excluding output)
"""

from typing import List

class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()

        min_diff = min(arr[i] - arr[i - 1] for i in range(1, len(arr)))

        ans = []

        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] == min_diff:
                ans.append([arr[i - 1], arr[i]])

        return ans

File: LeetCode/Sorting/1200-Minimum-Absolute-Difference.js
/**
 * Problem: 1200. Minimum Absolute Difference
 * Platform: LeetCode
 * Difficulty: Easy
 *
 * Approach:
 * 1. Sort the array.
 * 2. Find the minimum adjacent difference.
 * 3. Collect all adjacent pairs with that difference.
 *
 * Time Complexity: O(n log n)
 * Space Complexity: O(1) (excluding output)
 */

/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a, b) => a - b);

    let minDiff = Infinity;

    for (let i = 1; i < arr.length; i++) {
        minDiff = Math.min(minDiff, arr[i] - arr[i - 1]);
    }

    const ans = [];

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] - arr[i - 1] === minDiff) {
            ans.push([arr[i - 1], arr[i]]);
        }
    }

    return ans;
};